targets/csharp/source/source/PlayFabGameserverHttp.cs

```diff
@@ -0,0 +1,72 @@
+﻿// GSDK Only has the following supported frameworks for now
+#if NETSTANDARD2_0 || NETCOREAPP2_1
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace PlayFab
+{
+    // TODO: This should eventually be merged with PlayFabHttp
+    interface IGameserverHttpClient
+    {
+        Task<HeartbeatResponse> SendHeartbeatAsync(HeartbeatRequest request);
+    }
+
+    static class GameserverHttpClientFactory
+    {
+        public static IGameserverHttpClient Instance { get; set; }
```

Comment:
is this being set anywhere? Otherwise this should just be a get and the CreateInstance can just return this. We can create the value right here: Instance {get } = new GameServer*

Reply:
Shouldn't happen but if multiple threads call it, there could be two instances (not desirable)

---
In reply to: [255762196](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r255762196) [](ancestors = 255762196)

Reply:
It's set by the tests as a mock, we probably should consider moving to dependency injection instead like the rest of Control Plane. Since that's a larger change, I'll add a TODO for us to consider.

---
In reply to: [255765959](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r255765959) [](ancestors = 255765959,255762196)